compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3441,6 +3443,17 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
                 return getNextRuleForBindingPattern();
             case TUPLE_MEMBERS:
                 return ParserRuleContext.TUPLE_MEMBER;
+            case ON_FAIL_CHECK:
+                return ParserRuleContext.ON_KEYWORD;
+            case ON_KEYWORD:
+                return ParserRuleContext.FAIL_KEYWORD;
+            case FAIL_KEYWORD:
```

Comment:
We have this under `getNextRuleForKeywords()` too. Need do the change there. Not here.